code/src/java/plugin/pretokens/test/PreFactSetTester.java

```diff
@@ -72,26 +71,17 @@ public int passes(final Prerequisite prereq, final PlayerCharacter aPC, CDOMObje
 		return countedTotal(prereq, runningTotal);
 	}
 
-	private static <T> int getRunningTotal(final Prerequisite prereq, final int number, Iterable<Reducible> objModel,
+	private static <T> int getRunningTotal(final Prerequisite prereq, final int number, Collection<Reducible> objModel,
 		String factval, FactSetKey<T> fk)
 	{
 		T targetVal = fk.getFormatManager().convert(factval);
-		int runningTotal = 0;
-		CDO: for (Reducible r : objModel)
-		{
-			List<Indirect<T>> sets = r.getCDOMObject().getSetFor(fk);
-			for (Indirect<T> indirect : sets)
-			{
-				if (indirect.get().equals(targetVal))
-				{
-					runningTotal++;
-					continue CDO;
-				}
-			}
-		}
+		int runningTotal = (int) objModel.stream()
+		                                 .map(r -> r.getCDOMObject().getSetFor(fk))
+		                                 .filter(sets -> sets.stream()
```

Comment:
Wow, stream in a stream - is there anything here that can be broken out into a separate primitive or method to make it clearer?